lib/table_of_contents/parser.rb

```diff
@@ -13,13 +13,58 @@ def initialize(html)
       def build_toc
         toc = %Q{<ul class="section-nav">\n}
 
+        min_h_num = 6
         @entries.each do |entry|
-          toc << %Q{<li class="toc-entry toc-#{entry[:node_name]}"><a href="##{entry[:id]}#{entry[:uniq]}">#{entry[:text]}</a></li>\n}
+          h_num = entry[:node_name][1].to_i
+          min_h_num = [min_h_num, h_num].min
         end
+        toc << build_lis(@entries, min_h_num)
```

Comment:
Sometimes build_lis is called with a different parameter that isn't @entries. build_lis needs to know if it should use @entries or another list of entries, so it may as well receive @entries as a parameter.